RecoBTag/ONNXRuntime/plugins/DeepCombinedONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,345 @@
+  *(++ptr) = tag_info_features.trackSip2dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip2dSigAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dSigAboveCharm;
+  *(++ptr) = tag_info_features.jetNSelectedTracks;
+  *(++ptr) = tag_info_features.jetNTracksEtaRel;
+  assert(start + n_features_global_ - 1 == ptr);
+
+  // c_pf candidates
+  auto max_c_pf_n = std::min(features.c_pf_features.size(), (std::size_t)n_cpf_);
+  offset = i_jet * input_sizes_[kChargedCandidates];
+  for (std::size_t c_pf_n = 0; c_pf_n < max_c_pf_n; c_pf_n++) {
+    const auto& c_pf_features = features.c_pf_features[c_pf_n];
+    ptr = &data_[kChargedCandidates][offset + c_pf_n * n_features_cpf_];
+    start = ptr;
+    *ptr = c_pf_features.btagPf_trackEtaRel;
+    *(++ptr) = c_pf_features.btagPf_trackPtRel;
+    *(++ptr) = c_pf_features.btagPf_trackPPar;
+    *(++ptr) = c_pf_features.btagPf_trackDeltaR;
+    *(++ptr) = c_pf_features.btagPf_trackPParRatio;
+    *(++ptr) = c_pf_features.btagPf_trackSip2dVal;
+    *(++ptr) = c_pf_features.btagPf_trackSip2dSig;
+    *(++ptr) = c_pf_features.btagPf_trackSip3dVal;
+    *(++ptr) = c_pf_features.btagPf_trackSip3dSig;
+    *(++ptr) = c_pf_features.btagPf_trackJetDistVal;
+    *(++ptr) = c_pf_features.ptrel;
+    *(++ptr) = c_pf_features.drminsv;
+    *(++ptr) = c_pf_features.vtx_ass;
+    *(++ptr) = c_pf_features.puppiw;
+    *(++ptr) = c_pf_features.chi2;
+    *(++ptr) = c_pf_features.quality;
+    assert(start + n_features_cpf_ - 1 == ptr);
+  }
+
+  // n_pf candidates
+  auto max_n_pf_n = std::min(features.n_pf_features.size(), (std::size_t)n_npf_);
+  offset = i_jet * input_sizes_[kNeutralCandidates];
+  for (std::size_t n_pf_n = 0; n_pf_n < max_n_pf_n; n_pf_n++) {
+    const auto& n_pf_features = features.n_pf_features[n_pf_n];
+    ptr = &data_[kNeutralCandidates][offset + n_pf_n * n_features_npf_];
+    start = ptr;
+    *ptr = n_pf_features.ptrel;
+    *(++ptr) = n_pf_features.deltaR;
+    *(++ptr) = n_pf_features.isGamma;
+    *(++ptr) = n_pf_features.hadFrac;
+    *(++ptr) = n_pf_features.drminsv;
+    *(++ptr) = n_pf_features.puppiw;
+    assert(start + n_features_npf_ - 1 == ptr);
+  }
+
+  // sv candidates
+  auto max_sv_n = std::min(features.sv_features.size(), (std::size_t)n_sv_);
+  offset = i_jet * input_sizes_[kVertices];
+  for (std::size_t sv_n = 0; sv_n < max_sv_n; sv_n++) {
+    const auto& sv_features = features.sv_features[sv_n];
+    ptr = &data_[kVertices][offset + sv_n * n_features_sv_];
+    start = ptr;
+    *ptr = sv_features.pt;
+    *(++ptr) = sv_features.deltaR;
+    *(++ptr) = sv_features.mass;
+    *(++ptr) = sv_features.ntracks;
+    *(++ptr) = sv_features.chi2;
+    *(++ptr) = sv_features.normchi2;
+    *(++ptr) = sv_features.dxy;
+    *(++ptr) = sv_features.dxysig;
+    *(++ptr) = sv_features.d3d;
+    *(++ptr) = sv_features.d3dsig;
+    *(++ptr) = sv_features.costhetasvpv;
+    *(++ptr) = sv_features.enratio;
```

Comment:
this could also be a separate function in tensor_filler.cc, later to be integrated in DeepFlavour